Tests/TuistServerTests/Services/AuthTokenRefreshServiceTests.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import Mockable
+import Testing
+import TuistSupport
+
+@testable import TuistServer
+
+struct AuthTokenRefreshServiceTests {
+    private let subject: AuthTokenRefreshService
+    private let refreshAuthTokenService = MockRefreshAuthTokenServicing()
+    private let serverCredentialsStore = MockServerCredentialsStoring()
+    private let serverAuthenticationController = MockServerAuthenticationControlling()
+
+    init() {
+        subject = AuthTokenRefreshService(
+            refreshAuthTokenService: refreshAuthTokenService,
+            serverCredentialsStore: serverCredentialsStore,
+            serverAuthenticationController: serverAuthenticationController
+        )
+    }
+
+    @Test func stores_new_tokens() async throws {
+        // Given
+        let path = try TemporaryDirectory(removeTreeOnDeinit: true).path
+        given(refreshAuthTokenService).refreshTokens(serverURL: .any, refreshToken: .value("token")).willReturn(.init(accessToken:
+            "new-access-token", refreshToken: "new-refresh-token"))
+        given(serverAuthenticationController).authenticationToken(serverURL: .any).willReturn(.user(
+            legacyToken: nil,
+            accessToken: .test(),
+            refreshToken: .test()
+        ))
+        given(serverCredentialsStore)
+            .store(credentials: .any, serverURL: .any)
+            .willReturn()
+
+        // When
+        try await subject.refreshTokens(
+            path: path,
+            serverURL: URL(string: "https://test.tuist.io")!
```

Comment:
```suggestion
            serverURL: .test()
```

Nit: you can use the helper static factory instead.